### PR TITLE
Fix compatibility with GCC 10 and PGI 19.10

### LINKED ISSUE #83: CI failures
- AppVeyor fails on every configuration (seems a vcpkg building issue)
- Compilation error on MacOS - gcc@5 - openmpi configuration

In PR #82, MacOS - gcc@5 - openmpi configuration has been moved to allowed failures, while AppVeyor still fails. PGI compiler has been updated to 19.4 in commit 1d62d22.

### PR DESCRIPTION
**Pull request by titoiride**
Closes #83

### CODE REVIEW

**Comment by titoiride** [comment]
@cenit PGI configuration is now updated and the installation should be fine. However, different errors have now appeared in the various configurations:
- First one, there's a 
    ```CMake Error at C:/Tools/vcpkg/scripts/buildsystems/vcpkg.cmake:169 (_add_executable):
       Cannot find source file:
       C:/Program Files/Microsoft HPC Pack 2012/Inc/mpi.f90
    ```
    error, which I don't know how to fix... Is the MSMPI (HPC 2012, v4.1) broken?
- Second configuration, there is some linking error, but I cannot really understand the missing links

**Comment by cenit** [comment]
Moved this PR title to an updated target: PGI 19.10.
Let's start first from Windows, which is fully broken. We will move to 19.10 under linux/osx in a different PR

**Comment by titoiride** [comment]
🤩 By the way I was right now following the steps in [How to build](http://aladyn.github.io/ALaDyn/page/BUILD.html) for Windows on my laptop (powershell) to check if everything works. 
I I'd say that the most important problem is that CMake chooses automatically pgf95 while we need at least Fortran 2003+ standard. Is there a way to force CMake to pick pgfortran instead of pg95?
[Here](https://gitlab.kitware.com/cmake/cmake/blob/master/Modules/CMakeDetermineFortranCompiler.cmake#L88) it seems that the first one on the list is pgf95

**Comment by cenit** [comment]
looks like we have reached a terrible point:

pgfortran is incompatible with latest msmpi (https://github.com/microsoft/Microsoft-MPI/issues/3) while at the same time it crashes with an internal compiler error on gfortran 7.4 distributed by Cygwin (while it works perfectly with gfortran 7.4 and newer on mac and/or linux)

Let's try to find a better solution....

**Comment by titoiride** [comment]
@cenit Thank you for your really precious work! Given the latest Microsoft improvements towards an actual feasible Microsoft HPC implementation, I'd say to stick with the PGI compiler without recourring to Cygwin. Probably it is better to stay with native implementations rather than portings. Moreover, from other issues on the MSMPI pages, like [microsoft/Microsoft-MPI#33](https://github.com/microsoft/Microsoft-MPI/issues/33#issue-546397335) or [microsoft/Microsoft-MPI#28](https://github.com/microsoft/Microsoft-MPI/issues/28#issue-499373202), it seems that the Fortran modules of the MSMPI have problems of compatibility and implementation. I think (and hope) that they will fix them and maybe at that point the issue with the PGI compiler will be addressed!
As you said, for now PGI compiler is for debug reasons only and not for production. However, in a (probably very remote) future, it would be very interesting to use its accelerator capacity and run ALaDyn on a GPU!

**Comment by titoiride** [comment]
@cenit Could you write some small guide on how to generate a silent installer?
It was the most difficult part for me, I couldn't update it but with a trial-and-error and in the end I'm not even sure it was working...
I tried with some free software, but still I didn't understand. I know it may be tedious, but at least I can try by myself in the future (as long as we find a solution for the broken MPI)!

**Comment by cenit** [comment]
You need to create a "response file" for the instsller (InstallShield(

To create a response file, open a cmd prompt window as administrator then run:

```
pgiws64-1910.exe /r /f1"C:\\path\to\response\file\install.iss"
```

This will open a regular install with you needing to respond to the prompts with your responses stored in the "iss" file.

You can then use the "iss" file on other systems to perform an unattended install. Again, it's advised to run as administrator else Window will prompt you with a dialog box asking if you really want to allow the changes to your system.

To run in unattended mode, use the "/s" flag and the "/f1" flag with the full path to the response file.

```
pgiws64-1910.exe /s /f1"C:\\path\to\response\file\install.iss"
```` 


source:  https://www.pgroup.com/userforum/viewtopic.php?f=5&t=6352

**Comment by cenit** [comment]
@titoiride I launched a `fprettify --indent 1 -r src` on the source base.
I am also pending towards a quick merge of this PR soon. What's your opinion?

**Comment by titoiride** [comment]
@cenit I think that would be great! I've seen the prettify, I didn't know it but it seems very useful. I think I'll rerun it on the newer branches. What's the situation now? I've seen that on Cygwin the build succeeds, is that true? I've also seen you removed most of the tests on old compilers (I totally agree on that).
It's a shame that the PGI compiler doesn't work, that would be very useful to me since I'd like to run ALaDyn without using WSL on my laptop. This would allow me to use VS code and it's extensions without a double installation on the WSL. I hope the MPI issue will be fixed in the future releases!
Thank you again for all this work!

**Comment by cenit** [comment]
Wait a couple of days, it might work in the end :)

**Comment by cenit** [comment]
There is a promising new MPI release I am pushing on vcpkg. Also, I will restart discussions with msmpi authors to better find the problem

About cygwin, yes it works now.
On Windows, native, we can use now MSYS2 too!
Both are considered mandatory, while PGI can fail, on all platforms

About vscode: you use the remote extensions, am I right?

**Comment by titoiride** [comment]
That seems really good! Apart from ALaDyn, it think that it would be great to have a robust environment (linux like?) where to build and develop Fortran and HPC codes. I think Windows has really improved in the last years and working on it natively would be so useful!
Yes I'm using the WSL extension but, for example, I only installed the windows version of python that I use with the powershell (with the fortran linter) which works suprisingly well. So it would be fine to have everything under the same environment.

**Comment by cenit** [comment]
I hope that fortran becomes a first citizen in vcpkg. That would be huge, together with the flang reboot just happened in main llvm repository.
There is an effort at the moment running a little bit late, but I still have positive feelings for its success.

File: scripts/build/cmake.win.ps1 [hunk 1]
```diff
@@ -2,10 +2,10 @@
...
 $env:Path = "${env:PROGRAMFILES}/PGI/flexlm;$env:Path"
-$env:Path = "${env:PROGRAMFILES}/PGI/win64/19.4/bin;$env:Path"
+$env:Path = "${env:PROGRAMFILES}/PGI/win64/19.10/bin;$env:Path"
 $env:Path = "${env:Path};."
 $env:FLEXLM_BATCH = 1
-Write-Host "PGI 19.4 Enabled"
+Write-Host "PGI 19.10 Enabled"
 
 #Remove-Item .\build_win -Force -Recurse -ErrorAction SilentlyContinue
 New-Item -Path .\build_win -ItemType directory -Force
```
[the remark this answers is not in the document]
Comment (cenit) [reply scripts/build/cmake.win.ps1:11]: Ok, I will prepare something for that :)

**Comment by titoiride** [comment]
@cenit Sorry I took so long! I'm now merging this PR, I think that we can always open a new one if something must be updated again. Thank you very much for your support in this, I hope I can still reach you if I need some help 😄

**Comment by titoiride** [comment]
@cenit Just a last question: do you have any idea why now the MSYS toolchain is erroring with this?

```
:: Synchronizing package databases...
downloading mingw32.db...
downloading mingw32.db.sig...
error: mingw32: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" is unknown
:: Import PGP key 4096R/87771331B3F1FF5263856A6D974C8BE49078F532, "David Macek <david.macek.0@gmail.com>", created: 2018-01-14? [Y/n] error: mingw32: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: failed to update mingw32 (invalid or corrupted database (PGP signature))
downloading mingw64.db...
downloading mingw64.db.sig...
error: mingw64: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: failed to update mingw64 (invalid or corrupted database (PGP signature))
downloading msys.db...
downloading msys.db.sig...
error: msys: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: failed to update msys (invalid or corrupted database (PGP signature))
error: failed to synchronize all databases
Command exited with code 1
```

The macos gfortran is also erroring but I think that's because we should explicitly call `gfortran` among the packages to install. That configuration is always a pain...

**Comment by cenit** [comment]
MSYS just changed the basic installer and the old one broke. We already need another PR on this

**Pull request merged**